b2share/modules/schemas/cli.py

```diff
@@ -275,3 +282,78 @@ def community_schema_list_block_schema_versions(verbose, community, version=None
+    For an example of such format, execute 
+    b2share communities example_community_schema_part"""
+    comm = get_community_by_name_or_id(community)
+    if not comm:
+        raise click.BadParameter("There is no community by this name or ID: %s" 
+            % community)
+    if not os.path.isfile(json_file):
+        raise click.ClickException("%s does not exist on the filesystem" % 
+        json_file)
+    f = open(json_file,'r')
+    schema_dict = {}
+    try:
+        schema_dict = json.load(f)
+    except json.decoder.JSONDecodeError:
+        raise click.ClickException("%s is not valid JSON" % json_file)
+    f.close()
+    try:
+        validate_metadata_schema(schema_dict)
+    except:
+        raise click.ClickException("""%s is not a valid metadata schema for 
+        EUDAT community metadata format""" % json_file)
+    #create new block version schema
+    try:
+        community_schema = CommunitySchema.get_community_schema(comm.id)
+        _update_community_schema(comm, community_schema, schema_dict)
+    except CommunitySchemaDoesNotExistError:
+        _create_community_schema(comm, schema_dict)
+    db.session.commit()
+    click.secho("Succesfully processed new metadata schema")
+
+def _update_community_schema(community, community_schema, schema_dict):
+    comm_schema_json = json.loads(community_schema.community_schema)
+    if not('properties' in comm_schema_json.keys()):
+        raise click.ClickException("""Invalid community schema 
+            for community %s""" % community.name)
+    if not(len(comm_schema_json['properties'])==1):
+        raise click.ClickException("""Multiple block schemas not supported.""")
+    block_schema_id = comm_schema_json['properties'].popitem()[0]
+    block_schema = BlockSchema.get_block_schema(block_schema_id)
+    if block_schema is None:
+        raise ClickException("""CommunitySchema refers to block schema with id: %s . This schema does not exist""" % block_schema_id)
+    new_block_schema_version = block_schema.create_version(schema_dict)
+    block_schema_version_url= url_for(
+        'b2share_schemas.block_schema_versions_item',
+        schema_id = block_schema.id,
+        schema_version_nb = new_block_schema_version.version )
+    block_schema_version_url = block_schema_version_url + '#/json_schema'
+    comm_schema_json['properties'][str(block_schema.id)] = {}
+    comm_schema_json['properties'][str(block_schema.id)]['$ref'] = \
+        block_schema_version_url
+    new_cs = community_schema.create_version(community.id, comm_schema_json)
+
+def _create_community_schema(community, schema_dict):
+    community_schema = {'$schema':'http://json-schema.org/draft-04/schema#',
+            'properties':[""]}
+    block_schema = BlockSchema.create_block_schema(community.id, community.name)
+    bs_version = block_schema.create_version(schema_dict)
+    schema_link = block_schema_version_self_link(bs_version)
```

Comment:
Required: I think this is a bug, it should be `block_schema_version_json_schema_link`.